HLTrigger/JetMET/plugins/MVAJetPuIdProducer.cc

```diff
@@ -0,0 +1,205 @@
+         reco::Jet * corrJet = 0;
+         if( applyJec ) {
+             float scale = jec;
+                         corrJet = dynamic_cast<reco::Jet *>( jet.clone() );
+             corrJet->scaleEnergy(scale);
+         }
+         const reco::Jet * theJet = ( applyJec ? corrJet : &jet );
+         
+         PileupJetIdentifier puIdentifier;
+         if( produceJetIds_ ) {
+             puIdentifier = ialgo->computeIdVariables(theJet, jec,  &(*vtx), vertexes, runMvas_);
+             ids.push_back( puIdentifier );
+         } else {
+             puIdentifier = (*vmap)[jets.refAt(i)]; 
+             puIdentifier.jetPt(theJet->pt());    // make sure JEC is applied when computing the MVA
+             puIdentifier.jetEta(theJet->eta());
+             puIdentifier.jetPhi(theJet->phi());
+             ialgo->set(puIdentifier); 
+             puIdentifier = ialgo->computeMva();
+         }
+         
+         if( runMvas_ ) {
+             mvas[algoi->first].push_back( puIdentifier.mva() );
+             idflags[algoi->first].push_back( puIdentifier.idFlag() );
+             for( ++algoi; algoi!=algos_.end(); ++algoi) {
+                 ialgo = algoi->second;
+                 ialgo->set(puIdentifier);
+                 PileupJetIdentifier id = ialgo->computeMva();
+                 mvas[algoi->first].push_back( id.mva() );
+                 idflags[algoi->first].push_back( id.idFlag() );
+             }
+         }
+         
+         if( corrJet ) { delete corrJet; }
+     }
+     
+     if( runMvas_ ) {
+         for(vector<pair<string,MVAJetPuId *> >::iterator ialgo = algos_.begin(); ialgo!=algos_.end(); ++ialgo) {
+             vector<float> & mva = mvas[ialgo->first];
+             auto_ptr<ValueMap<float> > mvaout(new ValueMap<float>());
+             ValueMap<float>::Filler mvafiller(*mvaout);
+             mvafiller.insert(jetHandle,mva.begin(),mva.end());
+             mvafiller.fill();
+             iEvent.put(mvaout,ialgo->first+"Discriminant");
+             
+             vector<int> & idflag = idflags[ialgo->first];
+             auto_ptr<ValueMap<int> > idflagout(new ValueMap<int>());
+             ValueMap<int>::Filler idflagfiller(*idflagout);
+             idflagfiller.insert(jetHandle,idflag.begin(),idflag.end());
+             idflagfiller.fill();
+             iEvent.put(idflagout,ialgo->first+"Id");
+         }
+     }
+     if( produceJetIds_ ) {
+         assert( jetHandle->size() == ids.size() );
+         auto_ptr<ValueMap<StoredPileupJetIdentifier> > idsout(new ValueMap<StoredPileupJetIdentifier>());
+         ValueMap<StoredPileupJetIdentifier>::Filler idsfiller(*idsout);
+         idsfiller.insert(jetHandle,ids.begin(),ids.end());
+         idsfiller.fill();
+         iEvent.put(idsout);
+     }
+ }
+ 
+ 
+ 
+ void
+ MVAJetPuIdProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+         edm::ParameterSetDescription desc;
+     desc.setUnknown();
```

Comment:
This is not acceptable. The fillDescription method has to properly declare explicitly all parameters
which are read by the module plugin in its constructor and wherever/whatever else.